### PR TITLE
RCHAIN-3245 casper no warnings

### PR DESCRIPTION
**Pull request by rabbitonweb**
## Overview
Remove all compile warnings from casper.
Make compile warnings to act as compile errors.

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3245

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala [hunk 1]
```diff
@@ -46,7 +46,10 @@ class MultiParentCasperAddBlockSpec extends FlatSpec with Matchers with Inspecto
     val testProgram = for {
       deploy <- ConstructDeploy.basicDeployData[Effect](0)
       _      <- casper.deploy(deploy)
-      block  <- casper.createBlock.map { case Created(block) => block }
+      block <- casper.createBlock.map {
+                case Created(block) => block
+                case _              => throw new RuntimeException()
```
Comment (dzajkowski) [inline casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala]: that's not a helpful message.

File: casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala [hunk 2]
```diff
@@ -175,13 +178,17 @@ class MultiParentCasperAddBlockSpec extends FlatSpec with Matchers with Inspecto
                         .map {
                           case Created(block) =>
                             block.copy(sigAlgorithm = "invalid", sig = ByteString.EMPTY)
+                          case _ => throw new RuntimeException()
```
Comment (dzajkowski) [inline casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala]: same

File: casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala [hunk 3]
```diff
@@ -175,13 +178,17 @@ class MultiParentCasperAddBlockSpec extends FlatSpec with Matchers with Inspecto
...
                         }
 
       _ <- node0.casperEff.addBlock(unsignedBlock, ignoreDoppelgangerCheck[Effect])
       _ <- node1.transportLayerEff.clear(node1.local) //node1 misses this block
 
       signedBlock <- (node0.casperEff.deploy(data1) *> node0.casperEff.createBlock)
-                      .map { case Created(block) => block }
+                      .map {
+                        case Created(block) => block
+                        case _              => throw new RuntimeException()
```
Comment (dzajkowski) [inline casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala]: again

File: casper/src/test/scala/coop/rchain/casper/genesis/contracts/FailingResultCollectorSpec.scala [hunk 4]
```diff
@@ -18,6 +18,7 @@ class FailingResultCollectorSpec extends FlatSpec with AppendedClues with Matche
                   actual should not be (expected) withClue (clue(clueMsg, attempt))
                 case RhoAssertTrue(_, value, clueMsg) =>
                   value should not be (true) withClue (clue(clueMsg, attempt))
+                case _ => throw new RuntimeException()
```
Comment (dzajkowski) [inline casper/src/test/scala/coop/rchain/casper/genesis/contracts/FailingResultCollectorSpec.scala]: is this a new pattern?

File: casper/src/test/scala/coop/rchain/casper/helper/BlockGenerator.scala [hunk 5]
```diff
@@ -95,11 +95,12 @@ trait BlockGenerator {
       blockHash = Blake2b256.hash(header.toByteArray)
       body      = Body().withState(postState).withDeploys(deploys)
       serializedJustifications = justifications.toList.map {
-        case (creator: Validator, latestBlockHash: BlockHash) =>
-          Justification(creator, latestBlockHash)
+        case (cr: Validator, latestBlockHash: BlockHash) =>
```
Comment (dzajkowski) [inline casper/src/test/scala/coop/rchain/casper/helper/BlockGenerator.scala]: I strongly disagree with making names less obvious. creator is fine.

File: casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala (hunk 1 above)
Comment (rabbitonweb) [reply casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala]: tests

File: build.sbt [hunk 6]
```diff
@@ -147,6 +147,12 @@ lazy val casper = (project in file("casper"))
   .settings(inConfig(SlowcookerTest)(org.scalafmt.sbt.ScalafmtPlugin.scalafmtConfigSettings))
   .settings(
     name := "casper",
+    scalacOptions ++= Seq(
```
Comment (lukasz-golebiewski) [inline build.sbt]: This `Seq` should be extracted into a constant, as it's used in multiple sub-modules

File: casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala [hunk 7]
```diff
@@ -116,13 +117,19 @@ object InterpreterUtil {
               Log[F].warn(s"Found unknown failure") *> Right(none[StateHash])
                 .leftCast[BlockException]
                 .pure[F]
+            case UnusedCommEvent(_) => throw new RuntimeException("found UnusedCommEvent")
```
Comment (lukasz-golebiewski) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala]: You have a `Sync` constraint, so why not `raiseError`?

File: casper/src/test/scala/coop/rchain/casper/helper/BlockGenerator.scala (hunk 5 above)
Comment (rabbitonweb) [reply casper/src/test/scala/coop/rchain/casper/helper/BlockGenerator.scala]: `creator` is used a paramter in name, there was shadowing involved

File: casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala (hunk 7 above)
Comment (rabbitonweb) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala]: 👍

File: build.sbt (hunk 6 above)
Comment (rabbitonweb) [reply build.sbt]: 👍

File: casper/src/test/scala/coop/rchain/casper/genesis/contracts/TestUtil.scala [hunk 8]
```diff
@@ -101,25 +101,23 @@ object TestUtil {
...
       runtime: Runtime[F]
-  )(implicit context: ExecutionContext, rand: Blake2b512Random): F[Unit] =
+  )(implicit rand: Blake2b512Random): F[Unit] =
     ParBuilder[F].buildNormalizedTerm(code) >>= (evalTerm(_, runtime))
 
   private def evalTerm[F[_]: FlatMap](
       term: Par,
       runtime: Runtime[F]
-  )(implicit context: ExecutionContext, rand: Blake2b512Random): F[Unit] =
+  )(implicit rand: Blake2b512Random): F[Unit] =
```
Comment (KentShikama) [inline casper/src/test/scala/coop/rchain/casper/genesis/contracts/TestUtil.scala]: What was ExecutionContext used for?

@JosDenmark @ArturGajowy @tymm

Comment (rabbitonweb) [reply casper/src/test/scala/coop/rchain/casper/genesis/contracts/TestUtil.scala]: It was not used. Compiler was warning about it. Now it became an error.

**Pull request merged**